pkg/auth/iam/iam.go

```diff
@@ -60,6 +62,16 @@ type ErrorResponse struct {
 
 // NewFilter creates new Filter instance
 func NewFilter(client iam.Client) *Filter {
+	if s, exists := os.LookupEnv(constant.EnvSubdomainValidationEnabled); exists {
```

Comment:
Suggest to make it something like this below, 
So we could reuse it when we have another Filter option in the future:

```
        options := &FilterInitializationOptions{}        	
	if s, exists := os.LookupEnv(constant.EnvSubdomainValidationEnabled); exists {
		enableSubdomainFiltering, err := strconv.ParseBool(s)
		if err != nil {
			logrus.Errorf("Parse %s env error: %v", constant.EnvSubdomainValidationEnabled, err)
		}
		if enableSubdomainFiltering {
			options.AllowSubdomainMatchRefererHeaderValidation: true
		}
	}
	
	return &Filter{iamClient: client, options: options}
```

Reply:
updated